src/main/java/edu/cornell/kfs/sys/CUKFSKeyConstants.java

```diff
@@ -223,4 +223,11 @@
     public static final String ERROR_DOCUMENT_SUB_ACCOUNT_GLOBAL_CG_EXPENSE_REQUIRED_SECTION = "error.document.subAccountGlobal.cgExpenseRequiredSection";
     
     public static final String ERROR_DOCUMENT_CONTRACT_GRANT_INVOICE_PRORATE_NO_AWARD_BUDGET_TOTAL = "error.document.contractsGrantsInvoice.prorate.no.award.budget.total";
+    public static final String WARNING_CINV_FINAL_BILL_INDICATOR = "warning.document.contractGrantsInvoice.final.bill.indicator";
+    public static final String WARNING_CINV_BILLING_PERIOD_END_DATE_BEFORE_LAST_BILLED_DATE = "warning.document.contractGrantsInvoice.billingPeriodEndDate.before.lastBilledDate";
+    public static final String WARNING_CINV_BILLING_PERIOD_END_DATE_BEFORE_BILLING_PERIOD_START_DATE = "warning.document.contractGrantsInvoice.billingPeriodEndDate.before.billingPeriodStartDate";
+    public static final String WARNING_CINV_BILLING_PERIOD_START_DATE_BEFORE_LAST_BILLED_DATE = "warning.document.contractGrantsInvoice.billingPeriodStartDate.before.lastBilledDate";
+    public static final String WARNING_CINV_BILLING_PERIOD_START_DATE_BEFORE_AWARD_START_DATE = "warning.document.contractGrantsInvoice.billingPeriodStartDate.before.awardStartDate";
+    public static final String WARNING_CINV_DATE_RANGE_INVALID_FORMAT_LENGTH = "warning.document.contractGrantsInvoice.dateRange.expectedFormat";
```

Comment:
This constant's name doesn't quite match what would be expected for the given error key. I would recommend updating the constant name or error key accordingly for naming consistency.